### ISSUE TITLE
dose the plugin support ITF16

### ISSUE THREAD

**Issue by kw2019ltd** [DavBfr/dart_barcode#24]
hi,
dose the plugin support ITF16 ?  

![image](https://user-images.githubusercontent.com/47334522/107779707-6b4cbd00-6d4e-11eb-9ab7-8048a2e34619.png)

![image](https://user-images.githubusercontent.com/47334522/107779749-77d11580-6d4e-11eb-965e-b6b94b48a0e8.png)

**Comment by DavBfr** [comment]
I don't have this one. Do you have some documentation?

**Comment by kw2019ltd** [comment]
> 
> 
> I don't have this one. Do you have some documentation?

hi could not find exactly it seems less popular , i found below it seems its similar to ITF14 only it support 16 digit.

https://docs.esko.com/docs/en-us/dynamicbarcodes-for-ai/12/userguide/en-us/common/bar/reference/re_bar_ITF-16.html
https://www.han-soft.com/releases/barcode1d/documents/b_itf16.html

someone from UK send me attached image above and wrote to me : " ITF 16 is popular with amazon return labels which will see an increase in use"

if you can somehow add it and its not complex will be good.

**Comment by DavBfr** [comment]
It's implemented in the master branch.
Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).

**Comment by kw2019ltd** [comment]
> 
> 
> It's implemented in the master branch.
> Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).

sure thank you, i see build is failing can you have a look?

**Comment by DavBfr** [comment]
It's working now.

**Comment by kw2019ltd** [comment]
> 
> 
> It's working now.

thanks can you release new version? or you want me to tested first?

**Comment by DavBfr** [comment]
Yes, if you can test it first, that would be awesome.

In your pubspec.yaml, set:

```yaml
dependency_overrides:
  barcode:
    git:
      url: https://github.com/DavBfr/dart_barcode
      path: barcode
  barcode_widget:
    git:
      url: https://github.com/DavBfr/dart_barcode
      path: flutter
```

You'll need Flutter dev channel.

**Comment by kw2019ltd** [comment]
> 
> 
> Yes, if you can test it first, that would be awesome.
> 
> In your pubspec.yaml, set:
> 
> ```yaml
> dependency_overrides:
>   barcode:
>     git:
>       url: https://github.com/DavBfr/dart_barcode
>       path: barcode
>   barcode_widget:
>     git:
>       url: https://github.com/DavBfr/dart_barcode
>       path: flutter
> ```
> 
> You'll need Flutter dev channel.

not able to test it getting to many issues with dependency of flutter null check.

i see you migrate your code with null safety so not possible you can give a fix for old version without null safety

**Comment by kw2019ltd** [comment]
@DavBfr  hi after refracting you did in itf base class creating itf14 not working 

Barcode.itf14().isValid(text) return false , before it was working fine

![image](https://user-images.githubusercontent.com/47334522/109428677-c0e8c100-7a00-11eb-8a17-23a64bd1a5c3.png)

![image](https://user-images.githubusercontent.com/47334522/109428580-4ddf4a80-7a00-11eb-8216-65fdf5ea5872.png)



![image](https://user-images.githubusercontent.com/47334522/109428568-3ef89800-7a00-11eb-8e7e-904bea3b6c9e.png)

**Comment by kw2019ltd** [comment]
before retracting checksum calculated automatically 

![image](https://user-images.githubusercontent.com/47334522/109428806-5f752200-7a01-11eb-9f30-65f1f67e746b.png)

![image](https://user-images.githubusercontent.com/47334522/109428840-7c115a00-7a01-11eb-9c59-64c7d50e2f15.png)


![image](https://user-images.githubusercontent.com/47334522/109428819-6b60e400-7a01-11eb-8d2c-ec5bcce4bb0e.png)

**Comment by DavBfr** [comment]
Fixed in version 2.1.0

Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).
